src/jit/hwintrinsicxarch.cpp

```diff
@@ -262,14 +276,45 @@ GenTree* Compiler::impSSE41Intrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HA
 
 GenTree* Compiler::impSSE42Intrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* sig)
 {
+    GenTree*  retNode  = nullptr;
+    GenTree*  op1      = nullptr;
+    GenTree*  op2      = nullptr;
+    var_types callType = JITtype2varType(sig->retType);
+
+    CORINFO_ARG_LIST_HANDLE argLst = sig->args;
+    CORINFO_CLASS_HANDLE    argClass;
+    CorInfoType             corType;
     switch (intrinsic)
     {
         case NI_SSE42_IsSupported:
-            return gtNewIconNode(compSupports(InstructionSet_SSE42));
+            retNode = gtNewIconNode(compSupports(InstructionSet_SSE42));
+            break;
+
+        case NI_SSE42_Crc32:
+            assert(sig->numArgs == 2);
+            op2 = impPopStack().val;
+            op1 = impPopStack().val;
+#if defined(_TARGET_X86_) && !defined(_TARGET_AMD64_)
+            if (varTypeIsLong(callType))
+            {
+                return gtNewHelperCallNode(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, JITtype2varType(sig->retType));
+            }
+#endif
+            argLst  = info.compCompHnd->getArgNext(argLst);                        // the secnod argument
+            corType = strip(info.compCompHnd->getArgType(sig, argLst, &argClass)); // type of the secnod argument
+
+            retNode = gtNewScalarHWIntrinsicNode(callType, op1, op2, NI_SSE42_Crc32);
+
+            // TODO - currently we use the BaseType to bring the type of the secnod argument
+            // to the code generator. May encode the overload info in other way.
+            retNode->gtHWIntrinsic.gtBaseType = JITtype2varType(corType);
```

Comment:
I see, it's this that "enables" code gen to see a 16 bit version of CRC32.

Reply:
Yes, and 8 bye as well.